crates/bevy_reflect/src/path.rs

```diff
@@ -237,14 +243,14 @@ fn read_field<'r, 'p>(
     current: &'r dyn Reflect,
     field: &'p str,
     current_index: usize,
-) -> Result<&'r dyn Reflect, ReflectPathError<'p>> {
+) -> Result<&'r dyn Reflect, ReflectPathError> {
     match current.reflect_ref() {
         ReflectRef::Struct(reflect_struct) => {
             Ok(reflect_struct
                 .field(field)
                 .ok_or(ReflectPathError::InvalidField {
                     index: current_index,
-                    field,
+                    field: field.to_owned().into(), // possibly removable to_owned() ?
```

Comment:
.to_owned() is not removable here, so long as we're going with 'static `Cow`s, right?
```suggestion
                    field: field.to_owned().into(),
```